apps/remote_control/lib/lexical/remote_control/build/error.ex

```diff
@@ -16,15 +16,74 @@ defmodule Lexical.RemoteControl.Build.Error do
   end
 
   def parse_error_to_diagnostics(context, message_info, token) do
-    diagnostic = %Diagnostic{
+    if missing_terminator?(message_info) do
```

Comment:
In other words, why do we need this if? Why can't we just do what `build_missing_terminator_diagnostics` does?
```elixir
parse_error_diagnostic_functions = [&build_end_line_diagnostic/3, &build_start_line_diagnostic/3, &build_hint_diagnostic/3, &build_fallback_diagnostic/3]

Enum.flat_map(parse_error_diagnostic_functions, & &1.(context, message_info, token)
```
The only change is that each function needs to take 3 args and returns an empty list if it can't make a diagnostic.